packages/frontapp/src/app/components/CategorySearch/index.tsx

```diff
@@ -50,21 +55,106 @@ const StyledCheckBox = styled.input`
   width: ${dimensions.lg};
   height: ${dimensions.lg};
 `;
+
 type paramsType = {
   [key: string]: string[];
 };
+
 type subItemType = {
   title: string;
   id: number;
   checked: boolean;
+  numberOfItems?: number;
 };
+
 type itemsType = {
   title: string;
   inputType: string;
   id: number;
   subAttributes: subItemType[];
 };
+
 const CategorySearch = ({ setActive }: any) => {
+  let idOfItems = 0;
+  const numberOfInitialItems = 7; // number of items to show before clicked ShowMore
+  const [showMore, setShowMore] = useState(false);
+  const [allFilters, setAllFilters] = useState<itemsType[]>([]);
+  const { location } = useAppSelector((state) => state.user);
+  const [availableMaterial, setAvailableMaterial] = useState<any>([]);
+  const { data } = useGetAllMaterialsQuery({
+    variables: { location_id: location.id },
+    fetchPolicy: 'no-cache',
+  });
+
+  const allCategories = useMaterialFilter(availableMaterial, 'category');
+  const allAuthors = useMaterialFilter(availableMaterial, 'author');
+  const allTypes = useMaterialFilter(availableMaterial, 'type');
+  const allAvailability = useMaterialFilter(availableMaterial, 'availability');
+  const allSortBy = {
+    'By date added': undefined,
+    'By date of writing': undefined,
+  };
+
+  const customObjectFilter = (filterName: {
+    [author: string]: number | undefined;
```

Comment:
it might be better to use `?` character instead of `undefined`

Reply:
it screams at me, if i put ? anywhere in this piece of code

Reply:
okay